doc/maintenance/development_versions.rst

```diff
@@ -0,0 +1,46 @@
+.. _development_versions:
+
+How to Maintain Package Versions
+================================
+
+Sometimes, typically when proposing a change (e.g., modifying the Backend
+↔ Frontend protocol, or adding new methods into the
+``python3-copr-common`` package that other packages depend on), we need to
+specify version constraints between our components to avoid
+misinstallations.  Below are a few documented scenarios.
+
+Changing the Frontend ↔ Backend Protocol
+----------------------------------------
+
+There are two relevant values: ``MIN_FE_BE_API`` (backend side) and
+``Copr-FE-BE-API-Version`` (frontend side).
+
+Whenever the backend detects that the Frontend counterpart is older than
+expected, it delays processing.  Please bump the value anytime you think
+it's warranted—there's no risk in doing so, but forgetting to update it is
+risky (backend would process incompatible requests).
+
+To ensure smooth upgrades, always update the ``copr-backend`` machine
+first, so it is ready (after re-deployment) and waiting for the updated &&
+fully-compatible Frontend.
+
+Changes in python3-copr-common
+------------------------------
+
+This package is heavily used by all other components, except for
+`python3-copr` (which is intentionally self-contained).  Bumping its
+version is not a problem, since other components depend on it with
+a `>=` constraint.
+
+While we normally ship ``MAJOR.MINOR`` versions of our components, during
+development we may need to add a third ``.PATCH`` version component (this
+can be done arbitrarily in any pull-request — later, when making a
+release, we remove the ``.PATCH`` suffix and bump to either ``MAJOR+1`` or
+``MAJOR.MINOR+1``).
+
+Anytime you bump the ``.PATCH`` version, please update the corresponding
+``%copr_common_version`` macro in dependent components.
+
+Handling suffixes like ``dev1`` or ``1post`` is tricky, because RPM and
+Python (our main development language stack) compare such versions
+differently.
```

Comment:
This is still not enough.  We have automatically added `.post1` with tito, which breaks frontend:
```
No match for argument: python3dist(copr-common) >= 1.1.2
```
Because we provide:
```
$ dnf repoquery --disablerepo='*' --enablerepo hell --repofrompath=hell,https://download.copr.fedorainfracloud.org/results/@copr/copr-dev/fedora-43-x86_64/ --provides python3-copr-common-0:1.1.2.post1-1.git.107.08340e6.fc43.noarch
Updating and loading repositories:
Repositories loaded.
python-copr-common = 1.1.2.post1-1.git.107.08340e6.fc43
python3-copr-common = 1.1.2.post1-1.git.107.08340e6.fc43
python3.14-copr-common = 1.1.2.post1-1.git.107.08340e6.fc43
python3.14dist(copr-common) = 1.1
python3dist(copr-common) = 1.1
```